src/averageAges.js

```diff
@@ -37,7 +43,19 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const woman = people.filter(person => person.sex === 'f');
+
+  const womenFilter = withChildren
+    ? people.filter(person => people.find(child =>
+      child.mother === person.name))
+    : woman;
```

Comment:
```suggestion
 const womenFilter = withChildren
    ? people.filter(person => people.find(child =>
      child.mother === person.name))
    : people.filter(person => person.sex === 'f');
```

Reply:
if `withChildren` will be true you will have no need to do your filter in 46 line